src/runtime/string.go

```diff
@@ -89,6 +89,21 @@ func stringToBytes(x _string) (slice struct {
 	return
 }
 
+// Convert a string to []rune slice.
+func stringToRunes(s string) []rune {
+	var n = 0
+	for range s {
```

Comment:
Seems like it would be better to use `n := len(s)` here instead of a `range` loop, which would replace lines 94-99.

Reply:
No. Iterating over a string actually decodes utf-8, while calling `len` returns the length (in bytes). What we need here are the number of Unicode code points, not the number of bytes.
See this playground for a demonstration of the difference: https://play.golang.org/p/7Nn87fbtPwU

Reply:
OK great, I stand corrected.